app/assets/javascripts/application.js

```diff
@@ -19,3 +19,11 @@
 //= require multiple-select
 //= require jquery.maskedinput
 //= require_tree .
+$(function(){
+  $('form').on('change', '#date_for_later', function(){
+    if(this.value.length > 0){
```

Comment:
A few things here:

1. It's best to avoid putting javascript in the application.js file. Since this pertains to a specific view, let's create another file for it and put this there. Keep in mind that this will be CoffeeScript.
2. Prefer `$(this).val()` to `this.value`. In general, prefer `$(this)` in jQuery, for reasons which are complex but I can go into if you like.
3. For removing or adding a class based on a boolean value, there's always `toggleClass`, which with 1 argument will toggle it based on its present value, but with 2 will use the boolean argument to determine whether the class should be present or not.

All that to say, we end up with:

```coffee
$('form').on 'change', '#date_for_later', ->
  $('#notification').toggleClass 'hidden', $(this).val().length > 0
  return
```